tests/e2e/fake-plugins/polylang-test-plugin-rest-features/polylang-test-plugin-rest-features.php

```diff
@@ -0,0 +1,263 @@
+				'permission_callback' => '__return_true',
+				'args'                => array(
+					'slug' => array(
+						'description'       => 'Slug of the deleted language. If none is provided, all languages are deleted',
+						'type'              => 'string',
+						'required'          => false,
+					),
+				),
+			),
+		)
+	);
+
+	register_rest_route(
+		PLL_TEST_NAMESPACE,
+		'/options',
+		array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => 'pll_test_get_options',
+				'permission_callback' => '__return_true',
+			),
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => 'pll_test_edit_options',
+				'permission_callback' => '__return_true',
+				'args'                => array(
+					'key' => array(
+						'description'       => 'Option key.',
+						'type'              => 'string',
+						'required'          => true,
+					),
+					'value' => array(
+						'description'       => 'Option value.',
+						'type'              => 'string',
+						'required'          => true,
+					),
+				),
+			),
+			array(
+				'methods'             => WP_REST_Server::DELETABLE,
+				'callback'            => 'pll_test_delete_options',
+				'permission_callback' => '__return_true',
+			),
+		)
+	);
+}
+add_action( 'rest_api_init', 'pll_test_register_rest_routes' );
+
+/**
+ * Returns one or all languages.
+ *
+ * @since 1.0
+ *
+ * @param WP_REST_Request $request Current REST request.
+ * @return array Language object as array, or array of languages.
+ */
+function pll_test_get_languages( $request ) {
+	$slug = $request->get_param( 'slug' );
+
+	if ( ! empty( $slug ) ) {
+		$language = PLL()->model->get_language( $slug );
+
+		return $language->to_array();
+	}
+
+	$languages = PLL()->model->get_languages_list();
+	array_walk(
+		$languages,
+		function( $language ) {
+			return $language->to_array();
+		}
+	);
+
+	return $languages;
+}
+
+/**
+ * Creates a languages.
+ *
+ * @since 1.0
+ *
+ * @param WP_REST_Request $request Current REST request.
+ * @return array|WP_Error Array of language props on success, `WP_Error` on failure.
+ */
+function pll_test_create_languages( $request ) {
+	$locale               = $request->get_param( 'locale' );
+	$languages            = require POLYLANG_DIR . '/settings/languages.php';
+	$values               = $languages[ $locale ];
+	$values['slug']       = $values['code'];
+	$values['rtl']        = (int) ( 'rtl' === $values['dir'] );
+	$values['term_group'] = 0;
+	$admin_model          = new PLL_Admin_Model( PLL()->options );
+	$errors               = $admin_model->add_language( $values );
+
+	if ( is_wp_error( $errors ) ) {
+		return $errors;
+	}
+
+	$admin_model->clean_languages_cache();
+
+	return PLL()->model->get_language( $locale );
+}
+
+/**
+ * Deletes all languages.
+ *
+ * @since 1.0
+ *
+ * @return WP_Error|true `WP_Error` on failure, `true` on success.
+ */
+function pll_test_delete_languages() {
```

Comment:
According the args used when registering the route, this is supposed to receive an optional language slug and delete the related language.

Reply:
Indeed ... Fixed in 99c02441 😅